src/main/kotlin/it/unibo/alchemist/config/SimulationConfig.kt

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (C) 2010-2023, Danilo Pianini and contributors
+ * listed, for each module, in the respective subproject's build.gradle.kts file.
+ *
+ * This file is part of Alchemist, and is distributed under the terms of the
+ * GNU General Public License, with a linking exception,
+ * as described in the file LICENSE in the Alchemist distribution's top directory.
+ */
+
+package it.unibo.alchemist.config
+
+import it.unibo.alchemist.AlchemistExecutionOptions
+import it.unibo.alchemist.AlchemistExecutionOptions.Companion.defaultEndTime
+import it.unibo.alchemist.AlchemistExecutionOptions.Companion.defaultParallelism
+import it.unibo.alchemist.boundary.Loader
+import it.unibo.alchemist.config.SimulationConfig.SimulationConfigParameters.Companion.defaultEndTime
+import it.unibo.alchemist.config.SimulationConfig.SimulationConfigParameters.Companion.defaultParallelism
+import it.unibo.alchemist.config.VariablesParsingUtils.parse
+import it.unibo.alchemist.config.VariablesParsingUtils.parseBoolean
+import it.unibo.alchemist.config.VariablesParsingUtils.parseDouble
+import it.unibo.alchemist.config.VariablesParsingUtils.parseInt
+import it.unibo.alchemist.config.VariablesParsingUtils.parseMap
+import it.unibo.alchemist.config.VariablesParsingUtils.parseString
+import it.unibo.alchemist.config.VariablesParsingUtils.parseStringList
+
+/**
+ * Holds alchemist launch options configuration values.
+ *
+ * @property type launcher type
+ * @property parameters launcher parameters
+ */
+data class SimulationConfig(
+    val type: String = defaultLauncherType,
+    val parameters: SimulationConfigParameters,
+) {
+
+    /**
+     * @property variables selected batch variables. Defaults to [emptyList]
+     * @property parallelism parallel threads used for running locally. Defaults to [defaultParallelism]
+     * @property endTime final simulation time. Defaults to [defaultEndTime]
+     * @property isWeb true if the web renderer is used. Defaults to false.
+     * @property isBatch whether batch mode is selected.
+     * @property distributedConfigPath the path to the file with the load distribution configuration,
+     * or null if the run is local
+     * @property graphicsPath the path to the effects file, or null if unspecified
+     * @property serverConfigPath if launched as Alchemist grid node server,
+     * the path to the configuration file. Null otherwise.
+     * @property engineMode engine execution mode
+     * @property outputReplayStrategy batch engine mode output replay strategy
+     * @property batchSize batch engine mode fixed batch size
+     * @property epsilon epsilon engine mode epsilon threshold value
+     */
+    data class SimulationConfigParameters(
+        val variables: List<String>,
+        val parallelism: Int,
```

Comment:
depends on the launcher

Reply:
Removed, see general changes to Loader and Launcher creation